src/d3a/models/strategy/update_frequency.py

```diff
@@ -157,59 +156,32 @@ def time_for_price_update(self, strategy, time_slot):
         return self.elapsed_seconds(strategy) >= \
                self.update_interval.seconds * self.update_counter[time_slot]
 
-    def update_energy_price(self, market, strategy):
-        if market.id not in strategy.offers.open.values():
-            return
-
-        for offer, iterated_market_id in strategy.offers.open.items():
-            iterated_market = strategy.area.get_future_market_from_id(iterated_market_id)
-            if market is None or iterated_market is None or iterated_market.id != market.id:
-                continue
-            try:
-                iterated_market.delete_offer(offer.id)
-                updated_price = round(offer.energy * self.get_updated_rate(market.time_slot), 10)
-                new_offer = iterated_market.offer(
-                    updated_price,
-                    offer.energy,
-                    strategy.owner.name,
-                    original_offer_price=updated_price,
-                    seller_origin=offer.seller_origin,
-                    seller_origin_id=offer.seller_origin_id,
-                    seller_id=strategy.owner.uuid
-                )
-                strategy.offers.replace(offer, new_offer, iterated_market.id)
-            except MarketException:
-                continue
-
-    def update_market_cycle_offers(self, strategy):
-        for market in strategy.area.all_markets[:-1]:
-            self.update_counter[market.time_slot] = 0
-            self.update_energy_price(market, strategy)
+    def update_members(self, **kwargs):
+        for key, value in kwargs.items():
+            if hasattr(self, key):
+                setattr(self, key, value)
```

Comment:
Interesting approach, but unfortunately this is not foolproof because someone can easily break the operation of the class by assigning arguments like `number_of_available_updates` or `update_counter`. It would be much better if there were concrete arguments and an explicit assignment.